okhttp/src/main/kotlin/okhttp3/AsyncDns.kt

```diff
@@ -39,27 +39,39 @@ interface AsyncDns {
    */
   fun query(
     hostname: String,
+    originatingCall: Call?,
     callback: Callback,
   )
 
+  /** Returns a [Dns] that blocks until all async results are available. */
+  open fun asBlocking(): Dns {
```

Comment:
This all feels like a round-peg in a square-hole . . . building an async API and only ever using it in a blocking way.

I think we should build a regular `Dns` implementation for Android’s DNS stuff, and eject AsyncDns from our API until we’re ready to take advantage of it being async

Reply:
I was using it here, so you can have per-network Dns.  I don't think that's possible without the async android resolver API.

Reply:
Ejecting the AsyncDns API for now.  it's internal and moved to okhttp-android.